packages/commonwealth/client/vite.config.ts

```diff
@@ -0,0 +1,122 @@
+import react from '@vitejs/plugin-react-swc';
+import path from 'path';
+import { Alias, defineConfig, loadEnv } from 'vite';
+import { createHtmlPlugin } from 'vite-plugin-html';
+import tsconfigPaths from 'vite-tsconfig-paths';
+
+const projectRootDir = path.resolve(__dirname);
+
+function createScriptsResolver(folder: string): Alias {
+  const find = new RegExp(`^${folder}/(.*)$`);
+  return {
+    find,
+    replacement: path.resolve(projectRootDir, 'scripts', folder, '$1'),
+  };
+}
+
+// https://vitejs.dev/config/
+export default defineConfig(({ mode }) => {
+  const envPath = path.dirname(path.dirname(process.cwd())); // root project .env
+  const env = loadEnv(mode, envPath, '');
+  return {
+    root: projectRootDir,
+    plugins: [
+      react(),
+      createHtmlPlugin({
+        entry: `./scripts/index.tsx`,
+        template: './index.html',
+      }),
+      tsconfigPaths(),
+    ],
+    build: {
+      outDir: '../build',
+    },
+    server: {
+      port: 8080,
+      host: 'localhost',
+      proxy: {
+        '/api': {
+          target: env.BACKEND_PROXY_URL || 'http://localhost:3000',
+          changeOrigin: true,
+          secure: false,
+        },
+      },
+    },
+    resolve: {
+      alias: [
+        {
+          // matches only non-relative paths that end with .scss
+          find: /^([^.].*)\.scss$/,
+          replacement: path.resolve(projectRootDir, 'styles', '$1.scss'),
+        },
+        {
+          // resolves assets/
+          find: /^assets\/(.*)$/,
+          replacement: path.resolve(projectRootDir, 'assets', '$1'),
+        },
+        createScriptsResolver('hooks'),
+        createScriptsResolver('navigation'),
+        createScriptsResolver('state'),
+        createScriptsResolver('views'),
+        createScriptsResolver('controllers'),
+        createScriptsResolver('models'),
+        createScriptsResolver('helpers'),
+      ],
+      extensions: ['.ts', '.tsx', '.js', '.jsx'],
+    },
+    define: {
+      global: {},
```

Comment:
i think we can remove this

Reply:
This is needed for some crypto package that expects global to exist. We set it as an empty object to satisfy it.